### PR TITLE
[WIP] - managed field validation

### PR DESCRIPTION
**Pull request by ShaunYearStrong**

### CODE REVIEW

File: public/js/components/AtomEdit/CustomEditors/CTAEditor.js [hunk 1]
```diff
@@ -18,19 +18,19 @@ export class CTAEditor extends React.Component {
     return (
       <div className="editor editor-cta">
         <ManagedForm data={this.props.atom} updateData={this.props.onUpdate}>
-          <ManagedField fieldLocation="data.url" name="Link Url" isRequired={true}>
+          <ManagedField fieldLocation="data.url" name="Link Url" fieldId="url" isRequired={true}>
```
Comment (clloyd) [inline public/js/components/AtomEdit/CustomEditors/CTAEditor.js]: Does this fieldId need to be provided? Can we get a unique if from the react component (might not be human readable)

File: public/js/components/ManagedEditor/ManagedEditorForm.js [hunk 2]
```diff
@@ -12,5 +12,5 @@ export const ManagedForm = (props) => {
 
 ManagedForm.propTypes = {
   updateData: PropTypes.func.isRequired,
-  children: PropTypes.element.isRequired
+  children: PropTypes.array.isRequired
```
Comment (clloyd) [inline public/js/components/ManagedEditor/ManagedEditorForm.js]: I fixed this in my branch by setting to
```
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element)
  ])
```

File: public/js/util/validators.js [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
+
+
 export const isHttpsUrl = (value) => {
   const stringValue = typeof value === 'string' ? value : '';
-
   if (stringValue.match(/^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/)) {
     return true;
   } else {
-    return "Not a HTTPS url";
+    return {
```
Comment (clloyd) [inline public/js/util/validators.js]: :+1:

File: public/js/components/AtomEdit/CustomEditors/CTAEditor.js (hunk 1 above)
Comment (ShaunYearStrong) [inline public/js/components/AtomEdit/CustomEditors/CTAEditor.js]: that'd be fine. I'm not actually using it atm but thought a UID would be useful if we had a list of all field errors.

File: public/js/util/validateField.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    });
+  }
+
+  // Custom validators
+  if(customValidation) {
+    customValidation.forEach((validator) => {
+      const result = validator(fieldValue);
+
+      if (result !== true) {
+        errors.push(result);
```
Comment (clloyd) [inline public/js/util/validateField.js]: Do we want to throw an error here if it's not true... but also doesn't provide an object we recognise?

Comment (ShaunYearStrong) [inline public/js/util/validateField.js]: Good idea, it'd be good to catch it here rather that it being unrecognised on the field.

File: public/js/components/AtomEdit/CustomEditors/CTAEditor.js (hunk 1 above)
Comment (ShaunYearStrong) [inline public/js/components/AtomEdit/CustomEditors/CTAEditor.js]: @clloyd Hmm, looks like for now, the key is being set to `.0` for all of them, so it's not unique. This was more a change to fix the spaces in the `id`

**Pull request merged**